lib/dataContract/DataContractFactory.js

```diff
@@ -100,8 +100,8 @@ class DataContractFactory {
   createStateTransition(dataContract) {
     return new DataContractCreateTransition({
       protocolVersion: DataContract.PROTOCOL_VERSION,
-      dataContract: dataContract.toJSON(),
-      entropy: dataContract.getEntropy(),
+      dataContract: dataContract.toObject(),
+      entropy: dataContract.getEntropy().toBuffer(),
```

Comment:
I think it will work without the conversion.

Reply:
It works here, but there are some other cases when we call deep clone on such data, we'll get a fatal error in karma environment.